command/agent/agent.go

```diff
@@ -420,6 +421,15 @@ func convertServerConfig(agentConfig *Config) (*nomad.Config, error) {
 		conf.RPCMaxConnsPerClient = limit
 	}
 
+	// Set deployment rate limit
+	if rate := agentConfig.Server.DeploymentRateLimit; rate == 0 {
+		conf.DeploymentRateLimit = deploymentwatcher.LimitStateQueriesPerSecond
+	} else if rate > 0 {
+		conf.DeploymentRateLimit = rate
+	} else {
+		return nil, fmt.Errorf("deployment-rate-limit must be greater than 0")
+	}
+
```

Comment:
It seems like the parsing & error message needs to go here, but setting the default here requires a dependency on the deploymentwatcher for that constant, which doesn't seem right

Reply:
I think this is ok. Let's add a `Default` prefix to the const to make it obvious its no longer hardcoded throughout.

We should probably update https://github.com/hashicorp/nomad/blob/main/nomad/config.go#L380-L478 with that default as well.

Reply:
Added.